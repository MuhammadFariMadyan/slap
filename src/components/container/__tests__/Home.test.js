import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import "jest-styled-components";

import Connected, { Home } from "../Home";

describe("Home", () => {
  jest.mock("../ProjectListContainer", () => "ProjectListContainer");
  jest.mock("../../presentational/Button", () => "Button");

  const mockStore = configureMockStore();

  const setup = propOverrides => {
    const props = {
      createProject: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Home {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const store = mockStore({});
    const wrapper = shallow(<Connected store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders as connected", () => {
    const { wrapper } = setupConnected();

    expect(wrapper).toMatchSnapshot();
  });

  it("should map new to props", () => {
    const { wrapper } = setupConnected();

    expect(wrapper.props()).toMatchObject({
      createProject: expect.any(Function)
    });
  });

  it("should calls 'createProject' props", () => {
    const { wrapper, store } = setupConnected();

    wrapper.props().createProject();

    expect(store.getActions()).toEqual([
      {
        type: "ADD_PROJECT",
        payload: expect.any(Object)
      },
      {
        type: "ADD_SLIDE",
        payload: expect.any(Object)
      },
      {
        type: "ADD_CONTENT",
        payload: expect.any(Object)
      },
      {
        type: "SET_ACTIVE_PROJECT_ID",
        payload: expect.any(String)
      },
      {
        type: "SET_ACTIVE_SLIDE_ID",
        payload: expect.any(String)
      }
    ]);
  });
});
