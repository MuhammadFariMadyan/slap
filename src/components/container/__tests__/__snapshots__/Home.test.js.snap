// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should renders as connected 1`] = `
<Component
  createProject={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`Home should renders default 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Slap - Simplify Slides Creation
    </title>
  </HelmetWrapper>
  <styled.h1>
    Slap
  </styled.h1>
  <styled.p>
    Simplify Slides Creation
  </styled.p>
  <Styled(Button)
    caption="Add New Project"
    icon={<Styled(Component) />}
    onClick={[Function]}
  />
  <Connect(Component) />
</styled.div>
`;
